db/migrate/20211022131607_drop_cockpit.rb

```diff
@@ -0,0 +1,35 @@
+class DropCockpit < ActiveRecord::Migration[6.0]
+  class AssignedServerRole < ActiveRecord::Base; end
+
+  class ServerRole < ActiveRecord::Base; end
+
+  class SettingsChange < ActiveRecord::Base; end
+
+  class MiqUserRole < ActiveRecord::Base
+    has_and_belongs_to_many :miq_product_features,
+                            :join_table => "miq_roles_features",
+                            :class_name => "DropCockpit::MiqProductFeature"
+  end
+
+  class MiqProductFeature < ApplicationRecord
+    has_and_belongs_to_many :miq_user_roles,
+                            :join_table => "miq_roles_features",
+                            :class_name => "DropCockpit::MiqUserRole"
```

Comment:
habtm can cause problems in migrations because Rails creates these temporary models in the middle that don't play nice with cache clearing.  If we can get rid of these, that would be preferable, even though it might make the interface here clunky.

Reply:
You are suggesting creating a `MiqRoleFeature` model with no id?

Reply:
> You are suggesting creating a `MiqRoleFeature` model with no id?

Yes, create an MiqRoleFeature model and do the through join more manually. (note, rhe table should already have an id as it's a requirement)

Reply:
one I realized that the table still has an id, it was not so bad